### PR TITLE
member spotlight PR

### PR DESCRIPTION
**Pull request by eparubrub**

### CODE REVIEW

File: assets/styles/modernism.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
 :root {
   --color-brand: #cc6c13;
   --color-link: var(--color-brand);
+  --color-link-background: #cc6c1320;
```
Comment (tellaho) [inline assets/styles/modernism.css:4]: Wow, never have I ever seen a 8-digit hex code. Makes sense after reading into it (last FF -> 00 is alpha channel) and [coverage](https://caniuse.com/css-rrggbbaa) seems passable for us. 👍

File: assets/styles/modernism.css [hunk 2]
```diff
@@ -252,3 +253,9 @@ a.link:after {
     font-size: 18px;
   }
 }
+
+::-webkit-scrollbar {
```
Comment (tellaho) [inline assets/styles/modernism.css:257]: What's this actually do? Are you seeing unexpected scrollbars?

File: components/Spotlight/Spotlight.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    render(){
+        return (
+            <React.Fragment>
+                <SpotlightMain member={this.state.mainMember}/>
+                <SpotlightPreview members={members} selectMember={this.selectMember}/>
+            </React.Fragment>
+        )
+    };
+}
+
```
Comment (tellaho) [inline components/Spotlight/Spotlight.js:24]: I've always followed the loose convention to define presentational child components in the same file as the main/parent component. Less file hopping needed and keeps feature logic close to each other.

What do you think about moving `SpotlightMain` and `SpotlightPreview` to this file?

File: components/Spotlight/Spotlight.module.css [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+.MainCardTitle{
+    margin-top: -10px;
+    font-size: 2.2em;
+}
+
+.MainDescription{
+    max-height: inherit;
+    min-width: 300px;
+    width: 50%;
+    text-align: right;
```
Comment (tellaho) [inline components/Spotlight/Spotlight.module.css:32]: I see the aesthetic we're going for here, but you _generally_ want to left-align body copy. Anything else at length becomes hard to read over time. Which gets me wondering, how long are these going to be?

File: components/Spotlight/Spotlight.module.css [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+.MainDescriptionText{
+    font-size: 1.3em;
+    border-right: thick solid;
+    border-color: #cc6c13;
```
Comment (tellaho) [inline components/Spotlight/Spotlight.module.css:38]: Oh, you're one property short of using the border-right declaration to its fullest. You can use `border-right: thick solid var(--color-brand);` and kill line 38 altogether.

(Reminder: Let's use `var(--color-brand);` when we can.)

File: components/Spotlight/Spotlight.module.css [hunk 6]
```diff
@@ -0,0 +1,155 @@
+.MainSpotlightWrapper{
```
Comment (tellaho) [inline components/Spotlight/Spotlight.module.css:1]: CSS Modules seem neat! Maybe even cool. But the rest of our code uses this [CSS-in-JS convention](https://nextjs.org/docs/basic-features/built-in-css-support#css-in-js). For example, the [About Page](https://github.com/hawaiians/hawaiiansintech/blob/main/pages/about.js#L168-L230). 

Any strong opinions on one approach versus the other? I found that [CSS Modules](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css) is equally documented in the NextJS docs... so I'm either way. If not, I'd prefer we follow the current format (mostly to stay consistent).

File: components/Spotlight/Spotlight.module.css [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+.MainCard{
+    max-height: inherit;
+    width: 50%;
+    position: relative;
+}
+
+.MainCardTitle{
+    margin-top: -10px;
```
Comment (tellaho) [inline components/Spotlight/Spotlight.module.css:24]: I generally try to avoid negative margin values like the plague — they make for some tricky layout troubleshooting down the line.

What are you trying to achieve here? It looks like decreasing the vertical space above the title? I'd prefer we make the header narrower instead if this is the desired outcome.

File: components/Spotlight/SpotlightPreview.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React from "react"; 
+import styles from "./Spotlight.module.css"
+
+class SpotlightPreview extends React.Component{
+    render(){
+        return (
+            <table className="large tableContent" cellSpacing="0">
+                <thead id="tableHeader" Test>
```
Comment (tellaho) [inline components/Spotlight/SpotlightPreview.js:8]: Is this `thead` necessary?

File: components/Spotlight/SpotlightPreview.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import React from "react"; 
+import styles from "./Spotlight.module.css"
+
+class SpotlightPreview extends React.Component{
+    render(){
+        return (
+            <table className="large tableContent" cellSpacing="0">
```
Comment (tellaho) [inline components/Spotlight/SpotlightPreview.js:7]: In my experience, tables come with some really challenging side effects. I'm happy to relent on the use here since I suppose we're literally just building a table with two-columns per row... but past me is having flashbacks for bad times come future maintenance.

File: pages/spotlights.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import Head from "next/head";
+import Link from "next/link";
+import MetaTags from "../components/Metatags.js";
+import HitLogo from "../components/HitLogo.js";
+import Spotlight from "../components/Spotlight/Spotlight.js"
+import React from "react"; 
+
+
+export default function Home() {
```
Comment (tellaho) [inline pages/spotlights.js:9]: Oh, I'm noticing we're doing this wrong on every other page too now but we may be better off naming this function name to be representative of the page. I.e. swap `Home()` with `Spotlights()`.

I don't know when we will need it... I read [the docs](https://nextjs.org/docs/basic-features/pages) and nothing really talks about that function's name... so, I suppose you could leave this and no harm would happen. 🤷‍♂️

**Review by tellaho — CHANGES_REQUESTED** [review]
This progress looks great! Know you're learning React at the same time; this is an incredible first stab. I dropped some direct feedback on specific lines of code.

I'm dropping broader questions that could I'm thinking may drive overall feature direction below. **_But I'm super curious on your thoughts before you dive into any changes._**

### tl;dr
The current design feels uncertain on whether it wants to accommodate short-form or long-form content. The scrolls-within-scrolls, the truncated content, the mechanisms to flip through content. But a short form list feel very, very similar to what we already have built. 🤔 

If you were to ask, my gut is that the feature is better off supporting longer, editorial-like pieces on individuals. And that'd move us toward a more document-like architecture.

- **If Spotlights are going to have their own section, they should distinguish themselves from other features.** The current format doesn't accommodate for longer pieces, reinforced with the scroll-within-a-scroll anti-pattern (a [fascinating read](https://baymard.com/blog/inline-scroll-areas) if you're wanting to learn about the UX thinking on why it's _anti-_).

- Say, if short-form was the goal, or we're trying to manage how much we're really biting off to chew, my gut reaction is that we should consider "embellishing" the main list with highlighted individuals instead of creating _another filtered list of similar people_.

- One of the bigger drivers for long- versus short-form content. Who's writing?? LOL. Are you wanting to write? Will these be autobiographical? Should we seek out a writer? Or more fill-in-the-blank, displayed social media profile-esque?

File: assets/styles/modernism.css (hunk 1 above)
Comment (eparubrub) [reply assets/styles/modernism.css:4]: oh! thats a cool coverage site, I'll save that for future use I didn't even think about compatibility with certain ways of declaring colors

File: assets/styles/modernism.css (hunk 2 above)
Comment (eparubrub) [reply assets/styles/modernism.css:257]: oh shoot i totally goofed and forgot to remove this before comitting. I think I was trying to figure out an implementation where the inline scrolling was more distinguished so that users would easily tell that it was an inline scroll area. It was taking me too long to figure out so I abandoned the idea and settled with the current in-line scrolling (which after reading your feedback, [the inline-scrolling link you showed me](https://baymard.com/blog/inline-scroll-areas), and thinking about it, I agree, we should probably avoid inline scrolling overall and remove inline scrolling in general

File: components/Spotlight/Spotlight.js (hunk 3 above)
Comment (eparubrub) [reply components/Spotlight/Spotlight.js:24]: OH yea that makes sense. I didn't know you could have multiple components in one file, I'm guessing it would look something [like this](https://stackoverflow.com/a/50073244)? I see in this SO answer (and I think I remember writing it in my react notes) the react components are typically made 1:1 per file but like you said it logically makes sense to group them if one is just a presentational component for the main component. I agree, I think we should move `SpotlightMain` and `SpotlightPreview` to this file

File: components/Spotlight/Spotlight.module.css (hunk 6 above)
Comment (eparubrub) [reply components/Spotlight/Spotlight.module.css:1]: Nah no strong opinion on this. I only had a css module because it was helping my brain with debugging at the time. The example you linked in the about page seems pretty straightforward so I'm all for following the current format and changing it to the css-in-js convention

File: components/Spotlight/Spotlight.module.css (hunk 7 above)
Comment (eparubrub) [reply components/Spotlight/Spotlight.module.css:24]: Will keep that in mind, I'll make the change to take out negative margin vals. I think I was having trouble because there were some conflicting css elements and styles from the modernism css or from other elements we had from the navigation bar. It was just to bring the title a little bit higher for looks. Again, I can definitely remove it and debug from the element above or adjust the other elements accordingly

File: components/Spotlight/Spotlight.module.css (hunk 4 above)
Comment (eparubrub) [reply components/Spotlight/Spotlight.module.css:32]: got it, I set it to inherit the height from the MainSpotlight height (line 13 - 500px) in order to keep it constrained to the height of the main section (at least I believe thats what it's doing). After reading your overall feedback, I think we could definitely look into making this a lot longer to follow a more document-like structure so that it's not this small in-line scrolling text.

File: components/Spotlight/Spotlight.module.css (hunk 5 above)
Comment (eparubrub) [reply components/Spotlight/Spotlight.module.css:38]: oh right! sounds good thanks for that feedback and reminder

File: components/Spotlight/SpotlightPreview.js (hunk 9 above)
Comment (eparubrub) [reply components/Spotlight/SpotlightPreview.js:7]: haha I admit this was a pretty lazy implementation. A lot of the designs I went for on this page was playing on the safe side - trying to re-use current colors, ideas, and styles to keep a consistency in design. A by product of this thinking resulted in me re-using the tables and css from the home page.

We could definitely strip the table from the this. Do you think something like [grids](https://react-bootstrap.github.io/layout/grid/#row-layout-col-sizing) would be better? Or like you said in the overall feedback, maybe this would change if we moved toward another implementation where these wouldn't be present.

File: components/Spotlight/SpotlightPreview.js (hunk 8 above)
Comment (eparubrub) [reply components/Spotlight/SpotlightPreview.js:8]: Err I believe so, like i said in the above comment in line 7, this was a re-used implementation from the homepage, I think I tried playing around with it to make it work and `thread` was necessary, I can verify this if we decide to keep the implementation though. If it's not necessary, I can remove it

File: pages/spotlights.js (hunk 10 above)
Comment (eparubrub) [reply pages/spotlights.js:9]: Oh I didn't even realize that. Yea that's weird that react doesn't seem to care about what it's named but now that you pointed it out it's probably gonna bother me so I'd definitely want to correct this

**Comment by tellaho** [comment]
Closing for now braddah man. Let's circle back to this one soon.
